resources/widgetlist.ts

```diff
@@ -1,17 +1,36 @@
 // NOTE
 // This class should be considered deprecated, and any users of this class should
 // just switch over to using CSS grid.
+type Sorter = () => number;
 
 class WidgetList extends HTMLElement {
+  private _nextId = 1;
+  private _nameToId: { [key: string]: number } = {};
+  private _elements: { [key: number]: Sorter } = {};
+  private _sorted: number[] = [];
+  private _elementwidth = 100;
```

Comment:
Also, I think it's redundant to set all of these values here and then re-set them in attached.  Can these just be types and then the default values only be set in one place?